freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserService.java

```diff
@@ -112,39 +123,60 @@ private void asyncSynchronizeUsers(
         Set<String> userCrnFilter, Set<String> machineUserCrnFilter) {
         try {
             Set<String> environmentCrns = stacks.stream().map(Stack::getEnvironmentCrn).collect(Collectors.toSet());
+
+            Optional<String> requestIdOptional = Optional.of(
```

Comment:
why is it optional if it always has a value?

Reply:
This is eventually used in the calls to the GrpcUmsClient so that the UMS logs will contain the same requestId. The GrpcUmsClient method signatures take Optional<String>, so this is an optimization to avoid repetitively creating Optionals.
If you prefer, I could keep it as a String in this class and create an Optional in the UmsUsersStateProvider and the UmsEventGenerationIdsUtil instead. That would keep the UserService a little cleaner.